lang/rust/Portfile

```diff
@@ -166,7 +169,21 @@ set rust_platform       ${arch}-apple-${os.platform}
 set rust_root           ${worksrcpath}/build/stage0-${arch}
 
 if { ${subport} ne ${ccwrap} } {
+    if {${os.platform} eq "darwin" && ${os.major} < 13} {
+        depends_build-append \
+                        port:curl
+    }
+
     post-extract {
+        # Relink agains MacPort's libcurl allows to fix `error: Unknown option`
+        # which happened because cargo tries to use HTTP/2 and it fails
+        # See: https://github.com/rust-lang/cargo/blob/2e2a16e983f597da62bc132eb191bc3276d4b1bb/src/cargo/core/package.rs#L423
+        if {${os.platform} eq "darwin" && ${os.major} < 13} {
+            set sLib /usr/lib/libcurl.4.dylib
+            set lLib ${prefix}/lib/libcurl.4.dylib
```

Comment:
Minor point, but what happens here if curl is update in the future to no longer provide v4 dylib here. maybe make this just `${prefix}/lib/libcurl.dylib` ?

Reply:
fixed